typing/subst.ml

```diff
@@ -153,16 +182,24 @@ let rec typexp s ty =
             Tconstr(type_path s (Pdot(m,i',pos)), tl, ref Mnil)
         | _ -> assert false
       else match desc with
-      | Tconstr(p, tl, _abbrev) ->
-          Tconstr(type_path s p, List.map (typexp s) tl, ref Mnil)
+      | Tconstr (p, args, _abbrev) ->
+         let args = List.map (typexp s) args in
+         begin match PathMap.find p s.types with
+         | exception Not_found -> Tconstr(type_path s p, args, ref Mnil)
+         | Path _ -> Tconstr(type_path s p, args, ref Mnil)
```

Comment:
This call to `type_path` seems pointless since we already have the path as the argument to `Path _`.